### PR TITLE
sending canonical URL(s) along with user actions associated with a page

### PR DESCRIPTION
**Pull request by 2is10**
@leogrim @yingjieMiao @stkem 

Also no longer posting canonical URLs to `/ext/canonical` on every applicable page load.

Note that there is a `TODO` in eliza to propagate canonical URLs to shoebox. (I updated shoebox to optionally accept them from eliza.)

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/ExtMessagingController.scala [hunk 1]
```diff
@@ -47,12 +47,13 @@ class ExtMessagingController @Inject() (
       (o \ "text").as[String].trim,
       (o \ "recipients").as[Seq[String]])
 
-
+    // TODO: factor out NormalizedURI interning so that it's only done once instead of once per recipient
```
Comment (stkem) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/ExtMessagingController.scala]: Not sure what you mean here. We only make the call once per thread creation, not once per recipient.

Comment (2is10) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/ExtMessagingController.scala]: You’re right. I misread it. Removed the comment.

**Comment by 2is10** [comment]
I tried to close the loop in eliza, but message sending is a little too complicated. I don't understand why we need to optionally intern the URI in `MessagingController.sendMessage`. In the new thread case, it's already been done. In the reply case, the thread is already associated with a normalized URI. Maybe it's forward-looking code. Anyway, it's blocking me. @stkem I'll email you the patch I have so far.

**Comment by stkem** [comment]
I'd like to merge this, any objections?

**Comment by leogrim** [comment]
lgtm thanks Jared!

**Pull request merged**